### PR TITLE
Cirrus and Nox

### PR DESCRIPTION
**Pull request by trexfeathers**
This PR is an attempt to transfer the functionality of `travis.yml` into [`cirrus.yml`](https://github.com/SciTools-incubator/iris-ugrid/compare/master...trexfeathers:cirrus?expand=1#diff-53b8e6dd304c2a700c09db615dafa0b9de9b06a01d68b0c1355f0dadad511045) and [`noxfile.py`](https://github.com/SciTools-incubator/iris-ugrid/compare/master...trexfeathers:cirrus?expand=1#diff-f7a16a65f061822bcc73b8296f4dc837353d379d8d9cc5307982cb6941442835), replicating what has already been done with [SciTools-incubator/iris-esmf-regrid](https://github.com/SciTools-incubator/iris-esmf-regrid) and [SciTools/iris](https://github.com/SciTools/iris).

### CODE REVIEW

**Comment by trexfeathers** [comment]
Cirrus-Nox caching finally seems to be working 🎉

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [comment]: # (https://shields.io/ is a good source of these)
+[![Cirrus-CI]("https://api.cirrus-ci.com/github/SciTools-incubator/iris-ugrid.svg?branch=master")](https://cirrus-ci.com/github/SciTools-incubator/iris-ugrid)
```
Comment (alastair-gemmell) [inline README.md:2]: Do we need the quotation marks around https://....branch=master? The black line below doesn't have them. Might be a red herring but on the GitHub PR rich diff rendering of the new version doesn't display the cirrus-ci badge correctly

**Review by alastair-gemmell — APPROVED** [review]
Nice work @trexfeathers! This looks good to me. Only managed to spot a possible problem with the cirrus-ci badge in the README. But may be a red herring. See what you think. Happy to approve the PR.

File: README.md (hunk 1 above)
Comment (trexfeathers) [reply README.md:2]: I'll take care of that. But I suspect the bad rendering is because there hasn't yet been a Cirrus run against `master`

Comment (trexfeathers) [reply README.md]: 9242f3f16c2adf41c78c6a933c7d98668f677f75

**Pull request merged**